packages/react/src/components/TagButton/index.test.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import TagButton from './';
+import { IconType } from '../Icon';
+import axe from '../../axe';
+
+type TagButtonProps = {
+  icon: IconType;
+  label: string;
+  value: string;
+  onClick: () => void;
+};
+
+const defaultProps: TagButtonProps = {
+  icon: 'pencil',
+  label: 'Label: ',
+  value: 'value',
+  onClick: () => console.log('test')
+};
+
+const renderDefaultTagButton = (props = {}) => {
+  const mergedProps = { ...defaultProps, ...props };
+  return render(<TagButton {...mergedProps} />);
+};
+
+test('should render a button', () => {
+  renderDefaultTagButton();
+
+  expect(screen.getByRole('button')).toBeInTheDocument();
+  expect(screen.getByRole('button')).not.toHaveAttribute('role');
+  expect(screen.getByRole('button')).toHaveTextContent('Label: value');
+});
+
+test('should support className prop', () => {
+  renderDefaultTagButton({ className: 'foo' });
+
+  expect(screen.getByRole('button')).toHaveClass('foo');
```

Comment:
We should validate that it still includes the `TagButton` class:
```suggestion
  expect(screen.getByRole('button')).toHaveClass('TagButton', 'foo');
```